### ISSUE TITLE
Questions about sampling positive and negative training samples

### ISSUE THREAD

**Issue by yeliudev** [BigRedT/no_frills_hoi_det#7]
Many thanks for your great work! I have some questions about your sampling strategy of pos and neg samples.

1. According to `README.md`, using only the outputs of a pre-trained detector leads to a drop on recall of ground truth HOI labels (only 59% for the labelled case and 77% the unlabelled case) so that even if we ignore the object classes given by the detector, we can use only 77% of the training data to train the model. Did you manually add the missing training samples or just use the outputs from the detector to do the training?

2. In HICO-DET dataset, 'no_interaction' is treated as one kind of 'interaction' in the annotation file. However, most 'no_interaction' instances in the images are not annotated and this may lead to a huge drop when computing mAP. Did you label the negative samples of your training data as 'no_interaction' or 'none of the existing 600 HOI labels'?

Thank you again for your attention!

**Comment by BigRedT** [comment]
Great questions @goolhanrry!

1. Did you manually add the missing training samples or just use the outputs from the detector to do the training?
A: We only used the boxes returned by the object detector. So if a pair of ground truth boxes are not detected by the object detector we do miss out on the opportunity of training the model on that positive bbox pair. However, all images in the dataset are being used to at least get negatives if not positives. I agree that making use of all positives during training is an idea worth trying since it could improve the classifier even if those detections will be missed during test time (as a result of being missed by the object detector).   

2. Did you label the negative samples of your training data as 'no_interaction' or 'none of the existing 600 HOI labels'?
A: Each of the 600 HOI categories gets a binary label for any box-pair (since this is a multi-label setting as opposed to multi-class). Positives and negatives are thus defined with respect to each HOI category including HOI categories involving 'no-interaction'. Thus no-interaction is treated just like any other interactions (like 'riding' or 'sitting') with only box pairs annotated as 'no-interaction' being treated as positive for the corresponding HOI category.

Hope this answers your questions!

**Comment by yeliudev** [comment]
@BigRedT Thank you for your kind reply!
